apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/define/StatementEnhanceInfos.java

```diff
@@ -80,4 +87,19 @@ public void setParameter(int index, final Object parameter) {
     public int getMaxIndex() {
         return maxIndex;
     }
+
+    public Map<Integer, Object> getParameterMap() {
+        return parameterMap;
+    }
+
+    public List<Object> getSortParameters() {
+        List<Object> parameters = new ArrayList<Object>();
+        if (parameterMap.size() > 0) {
+            Iterator<Entry<Integer,Object>> iterator = parameterMap.entrySet().iterator();
+            while (iterator.hasNext()) {
+                parameters.add(iterator.next().getValue());
+            }
+        }
+        return parameters;
+    }
```

Comment:
There is still performance impact when using `TreeMap`, see [comment here](https://github.com/apache/skywalking/pull/2846#pullrequestreview-250196401), the performance is a big concern in agent side, please make a tradeoff